hydromt/workflows/region.py

```diff
@@ -0,0 +1,293 @@
+        filtered to only include basins with their outlet within the area of interest
+        ('outlets': true) of stream threshold arguments (e.g.: 'uparea': 1000).
+
+        Common use-cases include:
+
+        * {'basin': ID}
+
+        * {'basin': [ID1, ID2, ..]}
+
+        * {'basin': [x, y]}
+
+        * {'basin': [[x1, x2, ..], [y1, y2, ..]]}
+
+        * {'basin': /path/to/point_geometry}
+
+        * {'basin': [xmin, ymin, xmax, ymax]}
+
+        * {'basin': [xmin, ymin, xmax, ymax], 'outlets': true}
+
+        * {'basin': [xmin, ymin, xmax, ymax], '<variable>': threshold}
+
+        Subbasins are defined by its outlet locations and include all area upstream
+        from these points. The outlet locations can be passed as xy coordinate pairs,
+        but also derived from the most downstream cell(s) within a area of interest
+        defined by a bounding box or geometry, optionally refined by stream threshold
+        arguments.
+
+        The method can be speed up by providing an additional ``bounds`` argument which
+        should contain all upstream cell. If cells upstream of the subbasin are not
+        within the provide bounds a warning will be raised. Common use-cases include:
+
+        * {'subbasin': [x, y], '<variable>': threshold}
+
+        * {
+            'subbasin': [[x1, x2, ..], [y1, y2, ..]],
+            '<variable>': threshold, 'bounds': [xmin, ymin, xmax, ymax]
+            }
+
+        * {'subbasin': /path/to/point_geometry, '<variable>': threshold}
+
+        * {'subbasin': [xmin, ymin, xmax, ymax], '<variable>': threshold}
+
+        * {'subbasin': /path/to/polygon_geometry, '<variable>': threshold}
+
+        Interbasins are similar to subbasins but are bounded by a bounding box or
+        geometry and do not include all upstream area. Common use-cases include:
+
+        * {'interbasin': [xmin, ymin, xmax, ymax], '<variable>': threshold}
+
+        * {'interbasin': [xmin, ymin, xmax, ymax], 'xy': [x, y]}
+
+        * {'interbasin': /path/to/polygon_geometry, 'outlets': true}
+    """
+    kwargs = region.copy()
+    # NOTE: the order is important to prioritize the arguments
+    options = {
+        "basin": ["basid", "geom", "bbox", "xy"],
+        "subbasin": ["geom", "bbox", "xy"],
+        "interbasin": ["geom", "bbox", "xy"],
+        "geom": ["geom"],
+        "bbox": ["bbox"],
+        "mesh": ["UgridDataArray"],
+        "grid": ["raster"],
+    }
+    kind = next(iter(kwargs))  # first key of region
+    value0 = kwargs.pop(kind)
+
+    if kind in PLUGINS.model_plugins:
+        model_class = PLUGINS.model_plugins[kind]
+        other_model = model_class(root=value0, mode="r", logger=logger)
+        value0 = other_model.region
+        kwargs = dict(geom=other_model.region)
+        kind = "geom"
+    elif kind == "mesh":
+        if not _compat.HAS_XUGRID:
+            raise ImportError("xugrid is required to read mesh files.")
+
```

Comment:
Since #792 I beleive, xugrid has become a required dependency so this can go.

Reply:
pixi.toml and pyproject.toml tell us something else. As long as you don't import a MeshComponent it should be fine, right?

Reply:
Then we forgot to update those and should do so now